conmon-rs/server/src/child_reaper.rs

```diff
@@ -7,133 +7,88 @@ use log::{debug, error};
-                    Ok(status) => {
-                        if let WaitStatus::Exited(grandchild_pid, exit_status) = status {
-                            let grandchildren = Arc::clone(&grandchildren);
-                            thread::spawn(move || {
-                                if let Err(e) = Self::forget_grandchild(
-                                    grandchildren,
-                                    grandchild_pid,
-                                    exit_status,
-                                ) {
-                                    error!(
-                                        "Failed to reap grandchild for pid {}: {}",
-                                        grandchild_pid, e
-                                    );
-                                }
-                            });
-                        }
-                    }
-
-                    Err(err) => {
-                        // TODO FIXME this busy loops right now while there are no grandchildren.
-                        // There should be a broadcast mechanism so we only run this loop
-                        // while there are grandchildren
-                        if err != nix::errno::Errno::ECHILD {
-                            error!("caught error in reading for sigchld {}", err);
-                        }
-                    }
-                }
-            }
-        });
-        Ok(())
-    }
-
-    pub fn create_child<P, I, S>(&self, cmd: P, args: I) -> Result<std::process::ExitStatus>
+    pub async fn create_child<P, I, S>(&self, cmd: P, args: I) -> Result<ExitStatus>
     where
         P: AsRef<std::ffi::OsStr>,
         I: IntoIterator<Item = S>,
         S: AsRef<std::ffi::OsStr>,
     {
-        let mut cmd = std::process::Command::new(cmd);
+        let mut cmd = tokio::process::Command::new(cmd);
         cmd.args(args);
-
-        // To prevent an error: "No child processes (os error 10)",
-        // we prevent our reaper thread from calling waitpid until this child has completed.
-        // Inspired by
-        // https://github.com/kata-containers/kata-containers/blob/828a304883a4/src/agent/src/signal.rs#L27
-        let wait_lock = Arc::clone(&self.wait_lock);
-        let _lock = wait_lock
-            .lock()
-            .map_err(|e| format_err!("lock waitlock: {}", e))?;
-
         cmd.spawn()
             .map_err(|e| format_err!("spawn child process: {}", e))?
             .wait()
+            .await
             .map_err(|e| format_err!("wait for child process: {}", e))
     }
 
+    pub async fn exec_sync(
+        &self,
+        command: &Path,
+        args: Vec<String>,
+        timeout: i32,
+    ) -> Result<Output> {
+        let child = tokio::process::Command::new(command)
+            .stdout(Stdio::piped())
+            .stderr(Stdio::piped())
+            .kill_on_drop(true)
+            .args(args)
+            .spawn()
+            .map_err(|e| format_err!("spawn child process: {}", e))?;
```

Comment:
Let's replace those with the `anyhow::Context`:
```suggestion
            .context("spawn child process")?;
```

Reply:
fixed!